packages/block-editor/src/components/link-control/index.js

```diff
@@ -95,6 +99,7 @@ function LinkControl( {
 	settings,
 	onChange = noop,
 	showInitialSuggestions,
+	createEntity,
```

Comment:
> Can you explain the relationship between an "entity", a "value", and a "suggestion"

Here are my (admittedly flawed) definitions:

* `suggestion`- an option within the search results list which you can select but is not yet "submitted".
* `value` - the current selected value of the control. Will be one of the suggestions.
* `entity` - a resource created on demand which must conform to the `value` schema. Does not denote being in a selected state or otherwise. Simply a resource.

Looking harder at the above I wonder whether this is a terminology thing. I recognise that almost all the above can be normalised to all mean "suggestion". After all the `suggestion` has an object schema (set of properties it must have) and each of `value` and `entity` must conform to this. 

However, it felt strange to say I'm going to create a "suggestion", when you aren't creating a suggestion, you are creating a object which will become the selected value.

That said, I'm coming around to it being less confusing (if not perfect) if we simply normalise towards "suggestion". I'll rename accordingly.

Reply:
I've renamed to `createSuggestion`.

Reply:
> > Can you explain the relationship between an "entity", a "value", and a "suggestion"
> 
> Here are my (admittedly flawed) definitions:

I think how you've assessed it reads well to me. I've seen the overlap too, and have been conflicted on the extent to which a value _is_ a suggestion (and vice-versa). Some related discussion at https://github.com/WordPress/gutenberg/pull/19856#discussion_r371265123 .

Ultimately, I've come around to think it should be fine to treat `suggestion` as something which can become a value, or at least it includes an exhaustive set of properties which can be cherry-picked as desired to describe a possible value (so a `value` is a partial or complete set of properties from a suggestion).

I think entity _could_ fit a similar model, but I don't know that we need it. And just like I was stressed to try to eliminate a distinction between a value and a suggestion already, I wondered if we could do away with it here.

>I've renamed to `createSuggestion`.

It seems like a good change. I agree the wording feels strange to "create a suggestion", but I do think it's accurate to say that the prop is responsible for resolving to the thing which would be used as the suggestion object.